test/extended/cli/basics.go

```diff
@@ -250,6 +250,11 @@ var _ = g.Describe("[sig-cli] oc basics", func() {
 	})
 
 	g.It("can show correct whoami result with console", func() {
+		nsExist, err := exutil.IsNamespaceExist(oc, "openshift-config-managed")
```

Comment:
How is that namespace related to this test? Is that where the console runs? Or does the whoami command use information from that namespace?

Reply:
The console is an optional component now. What does `whoami` do if the namespace exists but the console URL is empty? Maybe there's a bug in `whoami` for this more extreme case where the console URL cannot be determined?

Reply:
it only is an issue if the "--show-console" arg is specified.  If you specify that arg when you don't have the console enabled, you'll get an error but that should not be surprising.

https://github.com/openshift/oc/blob/c078876cc5aa434a02f14c28b8c2a56b9011aa13/pkg/cli/whoami/whoami.go#L149-L150

https://github.com/openshift/oc/blob/c078876cc5aa434a02f14c28b8c2a56b9011aa13/pkg/cli/whoami/whoami.go#L121-L136

i feel like we had to deal w/ this for the "nocaps" e2e job, but since this test doesn't seem to contain any logic to check for which caps are enabled, i'm not sure how we ended up doing that.

Reply:
> Do we need another entry in the list in TestDecodeProvider for MicroShift?

No, that's just a test case for ensuring we detect/decode clusters correctly (so that we can pick the right set of tests).  I think you need to update the cluster discovery logic to recognize a "microshift" cluster and set the EnabledCapabilities accordingly:

https://github.com/openshift/origin/blob/5c4887ad5ff607fc70a51bcd8ccfb5f21502e0b0/cmd/openshift-tests/provider.go#L78
https://github.com/openshift/origin/blob/5c4887ad5ff607fc70a51bcd8ccfb5f21502e0b0/test/extended/util/cluster/cluster.go#L149-L153

Reply:
I've updated the [decodeProvider](https://github.com/openshift/origin/blob/5c4887ad5ff607fc70a51bcd8ccfb5f21502e0b0/cmd/openshift-tests/provider.go#L78) logic to recognize a "microshift" cluster and set `config.HasNoOptionalCapabilities = true`. However, the `[Skipped:NoOptionalCapabilities]` annotation doesn't seem to be getting enabled for this test. Any suggestions that I may be missing here?